lib/ttnt/tasks.rb

```diff
@@ -0,0 +1,54 @@
+require 'rake/testtask'
+require 'rake/file_list'
+require 'ttnt/testtask'
+require 'ttnt/test_selector'
+require 'rugged'
+
+# Reference: https://github.com/bundler/bundler/blob/master/lib/bundler/gem_helper.rb
+module TTNT
+  class TaskHelper
+    include Rake::DSL if defined? Rake::DSL
+
+    def self.install_tasks
+      new.install
+    end
+
+    def install
+      namespace :ttnt do
+        desc 'Generate test-to-code mapping for current commit object'
+        task 'anchor' do
+          # TODO: what if multiple test tasks are defined?
+          test_task = TTNT::TestTask.instances.first
+          test_files = []
+          test_files += test_task.test_files.to_a if test_task.test_files
+          test_files += Rake::FileList[test_task.pattern] if test_task.pattern
+
+          # TODO: properly regard run options defined for Rake::TestTask
+          gem_root = File.expand_path('..', File.dirname(File.expand_path(__FILE__)))
```

Comment:
Given we are on Ruby 2.2+, we can use:

~~~ruby
gem_root = File.expand_path('..', __dir__)
~~~